activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -105,7 +105,8 @@ def build(attributes = {}, &block)
         if attributes.is_a?(Array)
           attributes.collect { |attr| build(attr, &block) }
         else
-          add_to_target(build_record(attributes, &block))
+          record = build_record(attributes, &block)
```

Comment:
I think we can avoid this change

Reply:
it is my mistake. I reverted this place.